### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by fd0**
This is a collection of a lot of minor cleanups, each in its own commit.

### CODE REVIEW

File: archiver.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ const (
 var archiverAbortOnAllErrors = func(str string, fi os.FileInfo, err error) error { return err }
 var archiverAllowAllFiles = func(string, os.FileInfo) bool { return true }
 
+// Archiver is used to backup a set of directories.
```
Comment (fw42) [inline archiver.go]: Where does the Archiver get that set of directories from? it's stored in the `*server.Server` somehow? Maybe you can say "Archiver is used to back up a set of directories as references by a Server" or something like that?

File: archiver.go [hunk 2]
```diff
@@ -619,7 +623,9 @@ func (j archiveJob) Copy() pipe.Job {
 	return j.new
 }
 
-func (arch *Archiver) Snapshot(p *Progress, paths []string, pid backend.ID) (*Snapshot, backend.ID, error) {
+// Snapshot creates a snapshot of the given paths. If parentID is set, this is
+// used to compare the files with.
```
Comment (fw42) [inline archiver.go]: "compare the files with" sounds kinda weird. "compare the files to"?

File: backend/generic.go [hunk 3]
```diff
@@ -128,6 +128,8 @@ func (l *blobReader) Close() error {
 	return nil
 }
 
-func LimitReader(f io.ReadCloser, n int64) *blobReader {
-	return &blobReader{f: f, rd: io.LimitReader(f, n)}
+// LimitReadCloser returns a new reader wraps r in an io.LimitReader, but also
```
Comment (fw42) [inline backend/generic.go]: "new reader **that** wraps r"?

File: cache.go [hunk 4]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/restic/restic/server"
 )
 
+// Cache is used to handle the local cache.
```
Comment (fw42) [inline cache.go]: Local cache of what? What's stored in it?

File: cache.go [hunk 5]
```diff
@@ -51,6 +53,7 @@ func (c *Cache) Has(t backend.Type, subtype string, id backend.ID) (bool, error)
 	return true, nil
 }
 
+// Store returns an io.WriteCloser that is used to save new information to.
```
Comment (fw42) [inline cache.go]: "new information **to the cache**"?

File: cache.go [hunk 6]
```diff
@@ -73,6 +76,7 @@ func (c *Cache) Store(t backend.Type, subtype string, id backend.ID) (io.WriteCl
 	return file, nil
 }
 
+// Load returns information from the cache.
 func (c *Cache) Load(t backend.Type, subtype string, id backend.ID) (io.ReadCloser, error) {
```
Comment (fw42) [inline cache.go]: What about Close()ing of the WriteCloser and ReadCloser that are returned in all those methods? Does the caller have to close them? Is that important? What happens if I don't do that? Maybe add a comment "It's the caller's responsibility to close the returned reader" or something like that (if it's important)?

File: chunker/chunker.go [hunk 7]
```diff
@@ -52,7 +53,7 @@ func (c Chunk) Reader(r io.ReaderAt) io.Reader {
 	return io.NewSectionReader(r, int64(c.Start), int64(c.Length))
 }
 
-// A chunker internally holds everything needed to split content.
+// Chunker split content with Rabin Fingerprints.
```
Comment (fw42) [inline chunker/chunker.go]: s/split/splits/

**Comment by fw42** [comment]
This looks great, much clearer :heart:

File: archiver.go (hunk 1 above)
Comment (fd0) [inline archiver.go]: It's given to `Snapshot()`...

**Pull request merged**